crankshaft-monitor/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+//! Crate for monitoring crankshaft events.
+use std::net::SocketAddr;
+
+use proto::{Event, monitor_service_server::MonitorServiceServer};
+use server::CrankshaftMonitorServer;
+use tokio::sync::broadcast;
+
+pub mod proto;
+pub mod server;
+
+/// The main external Api to start the crankshaft monitor.
+pub async fn start_monitoring(
+    addr: SocketAddr,
+) -> Result<
+    (
+        broadcast::Sender<Event>,
+        tokio::task::JoinHandle<Result<(), tonic::transport::Error>>,
+    ),
+    Box<dyn std::error::Error>,
```

Comment:
This actually is pretty slick and works fine, but most of the time when you're passing around `Box<dyn std::error::Error>` you want to go ahead and instead be passing around an `anyhow::Error`. `anyhow` is already in the dependencies, so you should be able to just use it.